bundles/sirix-core/src/main/java/org/sirix/node/delegates/StructNodeDelegate.java

```diff
@@ -154,15 +197,22 @@ public void incrementChildCount() {
 
   @Override
   public int hashCode() {
-    return Objects.hashCode(childCount, nodeDelegate, firstChild, leftSibling, rightSibling, descendantCount);
+    return ((lastChild==-2L) ? Objects.hashCode(
+      childCount, nodeDelegate, firstChild, leftSibling, rightSibling, descendantCount) : Objects.hashCode(
+      childCount, nodeDelegate, firstChild, lastChild, leftSibling, rightSibling, descendantCount));
   }
 
   @Override
   public BigInteger computeHash() {
     final Funnel<StructNode> nodeFunnel = (StructNode node, PrimitiveSink into) ->
     {
-      into.putLong(node.getChildCount()).putLong(node.getDescendantCount()).putLong(node.getLeftSiblingKey()).putLong(
-          node.getRightSiblingKey()).putLong(node.getFirstChildKey());
+      if(lastChild != -2L) {
```

Comment:
Same as above. Maybe it should be a constant.

Reply:
Uh oh!! Will change